include/ibamr/CFGiesekusRelaxation.h

```diff
@@ -0,0 +1,134 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef included_CFGiesekusRelaxation
+#define included_CFGiesekusRelaxation
+/////////////////////////////// INCLUDES /////////////////////////////////////
+#include "IBAMR_config.h"
+
+#include "ibamr/CFRelaxationOperator.h"
+#include "ibamr/ibamr_enums.h"
+#include "ibamr/namespaces.h" // IWYU pragma: keep
+
+#include "Box.h"
+#include "CartesianPatchGeometry.h"
+#include "CellData.h"
+#include "CellIndex.h"
+#include "CellVariable.h"
+#include "Index.h"
+#include "IntVector.h"
+#include "Patch.h"
+#include "PatchGeometry.h"
+#include "PatchHierarchy.h"
+#include "PatchLevel.h"
+#include "Variable.h"
+#include "tbox/Database.h"
+#include "tbox/Pointer.h"
+#include "tbox/Utilities.h"
+
+#include <unsupported/Eigen/MatrixFunctions>
+
+#include <string>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class CFGiesekusRelaxation is a concrete CFRelaxationOperator that computes the relaxation function for the
+ * Giesekus fluid model.
+ */
+class CFGiesekusRelaxation : public IBAMR::CFRelaxationOperator
+{
+public:
+    /*!
+     * \brief This constructor reads in the parameters for the model from the input database.
+     */
+    CFGiesekusRelaxation(const std::string& object_name, SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db);
+
+    /*!
+     * \brief Empty destructor.
+     */
+    ~CFGiesekusRelaxation() = default;
+
+    /*!
+     * \name Methods to set patch data.
+     */
+    //\{
+
+    /*!
+     * \brief Evaluate the function on the patch interior.
+     */
+    void setDataOnPatch(const int data_idx,
+                        SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> var,
+                        SAMRAI::tbox::Pointer<SAMRAI::hier::Patch<NDIM>> patch,
+                        const double data_time,
+                        const bool initial_time = false,
+                        SAMRAI::tbox::Pointer<SAMRAI::hier::PatchLevel<NDIM>> patch_level =
+                            SAMRAI::tbox::Pointer<SAMRAI::hier::PatchLevel<NDIM>>(NULL)) override;
+
+    //\}
+
+private:
+    /*!
+     * \brief Default constructor.
+     *
+     * \note This constructor is not implemented and should not be used.
+     */
+    CFGiesekusRelaxation();
```

Comment:
`= delete` ?

Reply:
Yes, and it should be made `public`, see https://stackoverflow.com/a/18931192